packages/components/src/components/Table/Table.js

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Button,
+  DataTable,
+  DataTableSkeleton,
+  TableBatchAction,
+  TableBatchActions,
+  TableSelectAll,
+  TableSelectRow,
+  TableToolbar,
+  TableToolbarContent
+} from 'carbon-components-react';
+import { ALL_NAMESPACES } from '@tektoncd/dashboard-utils';
+
+import './Table.scss';
+
+const {
+  TableContainer,
+  Table: CarbonTable,
+  TableHead,
+  TableRow,
+  TableBody,
+  TableCell,
+  TableHeader
+} = DataTable;
+
+const Table = props => {
+  const {
+    batchActionButtons = [],
+    emptyTextAllNamespaces,
+    emptyTextSelectedNamespace,
+    headers: dataHeaders,
+    isSortable,
+    loading,
+    rows: dataRows,
+    selectedNamespace,
+    title,
+    toolbarButtons = []
+  } = props;
+
+  return (
+    <div className="tableComponent">
+      <DataTable
+        useZebraStyles
+        rows={dataRows}
+        headers={dataHeaders}
+        isSortable={isSortable}
+        render={({
+          rows,
+          headers,
+          getHeaderProps,
+          getRowProps,
+          getSelectionProps,
+          getBatchActionProps,
+          selectedRows
+        }) => (
+          <TableContainer title={title}>
+            <TableToolbar>
+              <TableBatchActions {...getBatchActionProps()}>
+                {batchActionButtons.map(button => (
+                  <TableBatchAction
+                    renderIcon={button.icon}
+                    key={`${button.text}Button`}
+                    onClick={() => {
+                      button.handler(
```

Comment:
Can we rename `handler` to `onClick` since it's a pretty common name for this and would likely be more familiar to other devs?